scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/rest/ScheduleRestController.java

```diff
@@ -35,7 +35,12 @@
 	public ResponseEntity<ApplicationScalingSchedules> getAllSchedules(@PathVariable String app_id) {
 		logger.info("Get All schedules for application: " + app_id);
 		ApplicationScalingSchedules savedApplicationSchedules = scheduleManager.getAllSchedules(app_id);
-
+		
+		// Any exception in getting the schedules
```

Comment:
My point is that it is all confused. It seems to be a guessing game when to catch an exception vs check the validationErrorResult. It isn't obvious since nothing is passed in or explicitly thrown.